src/ucp/core/ucp_ep.c

```diff
@@ -3565,3 +3567,31 @@ ucs_status_t ucp_ep_query(ucp_ep_h ep, ucp_ep_attr_t *attr)
 
     return UCS_OK;
 }
+
+ucs_status_t ucp_ep_realloc_slow_lanes(ucp_ep_h ep, uint32_t old_num_lanes, uint32_t new_num_lanes)
+{
+    ucp_lane_index_t lane;
+    int num_slow_lanes;
+
+    if (new_num_lanes <= UCP_MAX_FAST_PATH_LANES) {
+        ucs_free(ep->ext->uct_eps);
+        ep->ext->uct_eps = NULL;
+        return UCS_OK;
+    }
+
+    num_slow_lanes = new_num_lanes - UCP_MAX_FAST_PATH_LANES;
+
+    ep->ext->uct_eps = ucs_realloc(ep->ext->uct_eps,
+                                   num_slow_lanes * sizeof(uct_ep_h),
+                                   "ext_uct_eps");
+    if (ep->ext->uct_eps == NULL) {
+        return UCS_ERR_NO_MEMORY;
```

Comment:
need to release (or restore) previous array. or it will leak.

Reply:
ulresolved

Reply:
won't it be released in `ucp_ep_deallocate`?

Reply:
how, if ep_ext->uct_eps is NULL now?

Reply:
I added a temp var, you sure you see the latest commit?